From 9c0a2927692d2f0f0f0dd5370f622ad117886b55 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Tue, 12 Jun 2018 16:55:18 +0300
Subject: [PATCH 3980/5242] MLK-18583: Revert "ARM: dts: imx7d-sdb: Add DRM
 panel support"

commit  318e04958ffbc65860a1808a3d24b6922e674d20 from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit d8236af53074c61b7f88af773357f3cdcf378a9d.

The mxsfb driver uses old bindings so we need to provide those.

The LCD timings were already upstreamed but then an upstream commit
removed them in favor of adding a DRM path. Making mxsfb work requires
reverting this upstream commit

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts |   47 +++++++++++++++++++--------------------
 1 file changed, 23 insertions(+), 24 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 50a11aa..b2126c7 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -106,14 +106,6 @@
 		startup-delay-us = <200000>;
 	};
 
-	reg_lcd_3v3: regulator-lcd-3v3 {
-		compatible = "regulator-fixed";
-		regulator-name = "lcd-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&extended_io 7 GPIO_ACTIVE_LOW>;
-	};
-
 	reg_can2_3v3: regulator-can2-3v3 {
 		compatible = "regulator-fixed";
 		regulator-name = "can2-3v3";
@@ -144,19 +136,6 @@
 		enable-active-high;
 	};
 
-	panel {
-		compatible = "innolux,at043tn24";
-		pinctrl-0 = <&pinctrl_backlight>;
-		enable-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
-		power-supply = <&reg_lcd_3v3>;
-
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&display_out>;
-			};
-		};
-	};
-
 	backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm1 0 5000000>;
@@ -601,11 +580,31 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_lcdif>;
 	enable-gpio = <&extended_io 7 GPIO_ACTIVE_LOW>;
+	display = <&display0>;
 	status = "okay";
 
-	port {
-		display_out: endpoint {
-			remote-endpoint = <&panel_in>;
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+
+			timing0: timing0 {
+				clock-frequency = <9200000>;
+				hactive = <480>;
+				vactive = <272>;
+				hfront-porch = <8>;
+				hback-porch = <4>;
+				hsync-len = <41>;
+				vback-porch = <2>;
+				vfront-porch = <4>;
+				vsync-len = <10>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
 		};
 	};
 };
-- 
1.7.9.5

